crates/hyperqueue/src/server/event/log/stream.rs

```diff
@@ -68,6 +80,13 @@ async fn streaming_process(
                             break
                         }
                     }
+                    Some(EventStreamMessage::RegisterListener(tx)) => {
```

Comment:
How exactly does this work? It reads the events.. from the file system log file? And also blocks this `async fn`?

Reply:
Yes it blocks the thread, but only the thread that is responsible for writing events into the file. We need this to avoid modification of file while we are reading from it.

Reply:
I understand it now. But please add some comments about this to the code, it's not obvious how this works and that it depends on blocking the writer thread.